### ISSUE TITLE
Questions on Number 4

### ISSUE THREAD

**Issue by BMT45** [ebeshero/DHClass-Hub#554]
For number 4, I'm stuck on letter b because I can't figure out how to use the min and max functions to find the answer.  I tried //div//div[@type="assign"][contains(.,"GitHub" ) =>min()] to get the minimum value but I would still get the same number of results I would without the min function at the end of the contains function.

**Comment by ebeshero** [comment]
@BMT45 Ahh, this is a difficult one. Let's break it down a bit. You want to find the longest and the shortest divs (of the kind you successfully isolated). So, you need to start by getting a numerical measurement--literally the number of characters in the string. This is where you'd use the `string-length()` function. First, try just getting the `string-length()` on each of those divs--you can use simple map `!` for that.

**Comment by ebeshero** [comment]
@BMT45 Once you return the string-lengths, use the arrow operator to get the max and the min...

**Comment by ebeshero** [comment]
@BMT45 Return the numbers first so you know you're reaching the right nodes... Then, afterwards, you'll need to find a way to return the div element(s) that measure up to that max (and min) string-length. One step at a time.

**Comment by BMT45** [comment]
Yeah I got it thank you.  I also had a question for 3 b.  I tried using the concat expression //head//date ! [concat( '\d{4}', '\d{2}', '\d{2}')] to try to get a number that has 4 digits followed by a 2 digit number and that is followed by another number with 2 digits.  I know you need to use a concat function for this because you are looking for more than 2 arguments to answer the question.

**Comment by ebeshero** [comment]
@BMT45 ...I know you're primed to use those string functions, but this one (3b) doesn't require `concat()`.

**Comment by BMT45** [comment]
Would you just put "\d{4}-\d{2}-\d{2}" as the attribute value or could you put it in a contain() or match() function instead?  I just don't understand how to apply the attribute @when since the value is numbers that vary from entry to entry.

**Comment by ebeshero** [comment]
@BMT45 It's simpler than that--no regex required here. If you've found the Fridays on the syllabus, and you're returning their `<date>` elements, take one more step and take a look at the `@when` attribute. Notice that the values on that attribute are all formatted the same way...

**Comment by BMT45** [comment]
I got it thank you very much!

**Comment by ebeshero** [comment]
Yay! onward!
